app/components/Vault/index.js

```diff
@@ -1384,9 +1384,23 @@ const Vault = (props) => {
     );
 
   const minDecimals = setDecimals(decimals);
-  const showVaults =
-    (!vaultsToHide.includes(vault.address) || balanceOf >= 10 ** minDecimals) &&
+  const migrationAvailable = _.get(vault, 'migration.available');
+  const userHasBalance = balanceOf >= 10 ** minDecimals;
+  const vaultIsV1Type = vault.type === 'v1';
+  const vaultIsSpecial = vault.special;
+
+  let showVaults =
+    (!vaultsToHide.includes(vault.address) || userHasBalance) &&
     !hackedOrToBeAbsolutelyRemoved.includes(vault.address);
+
```

Comment:
It was necessary to add new logic to hide v1 and migratable vaults where the user has no balance